connectivity/FEATURE_BLE/include/ble/Gap.h

```diff
@@ -1435,29 +1431,36 @@ class Gap {
     GapShutdownCallbackChain_t &onShutdown();
 
 #if !defined(DOXYGEN_ONLY)
+    /*
+     * Constructor from the private implementation.
+     */
+    Gap(impl::Gap* impl) : impl(impl) {}
```

Comment:
I'm fine with this but I want to be on record being against an extra step of indirection, I don't believe making it look neater is worth the cost

Reply:
There are other options such as implementing things directly and using the private pointer for holding data but that was too much changes in one go. Private implementation makes us more robust regards to ABI changes and cleanly isolate from private headers which won't be accessible with the CMakeification.